libhoney/src/test/java/io/honeycomb/libhoney/End2EndBuilderTest.java

```diff
@@ -0,0 +1,361 @@
+import io.honeycomb.libhoney.LibHoney;
+import io.honeycomb.libhoney.ResponseObserver;
+import io.honeycomb.libhoney.TransportOptions;
+import io.honeycomb.libhoney.ValueSupplier;
+import io.honeycomb.libhoney.builders.HoneyClientBuilder;
+import io.honeycomb.libhoney.eventdata.EventData;
+import io.honeycomb.libhoney.responses.ClientRejected;
+import io.honeycomb.libhoney.responses.Response;
+import io.honeycomb.libhoney.responses.ServerAccepted;
+import io.honeycomb.libhoney.responses.ServerRejected;
+import io.honeycomb.libhoney.responses.ServerResponse;
+import io.honeycomb.libhoney.responses.Unknown;
+import org.apache.commons.lang3.StringUtils;
+import org.assertj.core.api.Condition;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.TimeUnit;
+
+import static com.github.tomakehurst.wiremock.client.WireMock.aResponse;
+import static com.github.tomakehurst.wiremock.client.WireMock.containing;
+import static com.github.tomakehurst.wiremock.client.WireMock.equalTo;
+import static com.github.tomakehurst.wiremock.client.WireMock.equalToJson;
+import static com.github.tomakehurst.wiremock.client.WireMock.matching;
+import static com.github.tomakehurst.wiremock.client.WireMock.matchingJsonPath;
+import static com.github.tomakehurst.wiremock.client.WireMock.post;
+import static com.github.tomakehurst.wiremock.client.WireMock.postRequestedFor;
+import static com.github.tomakehurst.wiremock.client.WireMock.stubFor;
+import static com.github.tomakehurst.wiremock.client.WireMock.urlPathMatching;
+import static com.github.tomakehurst.wiremock.client.WireMock.verify;
+import static io.honeycomb.libhoney.TransportOptions.DEFAULT_BATCH_SIZE;
+import static io.honeycomb.libhoney.TransportOptions.builder;
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * This ensures HoneyClientBuilder created clients match expected results as defined by {@link End2EndTest}.
+ */
+public class End2EndBuilderTest {
+
+    @Rule
+    public WireMockRule wireMock = new WireMockRule(8089);
+
+    private HoneyClient honeyClient;
+    private BlockingQueue<Response> notifyQueue;
+
+    @Before
+    public void setup() {
+        stubServer(200, "[" +
+            "  {\"status\": 202}" +
+            "]");
+    }
+
+    // this client will not timeout batches
+    private void createDefaultClient() throws URISyntaxException {
+        honeyClient = createBuilder().build();
+        notifyQueue = createObserverQueue();
+    }
+
+
+    private HoneyClientBuilder createBuilder() throws URISyntaxException {
+        return new HoneyClientBuilder().writeKey("testWriteKey").dataSet("testDataSet").apiHost("http://localhost:8089");
```

Comment:
Would rather `"http://localhost:8089"` be something like `LOCAL_TEST_URL` or similar.

Reply:
Done.